inc/Engine/Common/PerformanceHints/Database/Queries/AbstractQueries.php

```diff
@@ -60,6 +60,11 @@ public function delete_by_url( string $url ) {
 
 		$deleted = true;
 		foreach ( $items as $item ) {
+			if ( ! is_object( $item ) || ! isset( $item->id ) ) {
+				$deleted = false;
+				continue;
+			}
+
```

Comment:
I think this means that if an item in the array contains not valid element (element is not object or doesn't have id) we will bypass the current loop iteration after setting deleted with false, then we use this deleted with `&&` at the end we will have the final deleted for all upcoming items to be false, which will make `$this->delete_item( $item->id )` a dead zone (not reachable)

Is this what we expect here? even if this is an edge case I wanted ti highlight it.

Reply:
Yes, if the item is not a valid object and doesn't have id, we shouldn't get to `$this->delete_item( $item->id )` as that will throw and error since id is not present

Reply:
I don't think that I explained it well, I didn't mean the not valid item, I meant if we have an array of some valid items and some not valid items,in this case we will not call delete_item for all of them, is think expected?

Reply:
Yes, we would only call the delete_item for the valid ones. which is why the deleted is set to false in that logic before continue